crates/bench/benches/uv.rs

```diff
@@ -0,0 +1,112 @@
+//! Benchmarks uv using criterion.
+//!
+//! The benchmarks in this file assume that the `uv` and `virtualenv` executables are available.
+//!
+//! To set up the required environment, run:
+//!
+//! ```shell
+//! cargo build --release
+//! ./target/release/uv venv
+//! source .venv/bin/activate
+//! ./target/release/uv pip sync ./scripts/bench/requirements.txt
+//! ```
+
+use std::env;
+use std::path::Path;
+use std::process::{Command, Stdio};
+
+use bench::criterion::{criterion_group, criterion_main, measurement::WallTime, Criterion};
+use criterion::BatchSize;
+
+use fs_err as fs;
+
+const REQUIREMENTS: [&str; 4] = [
+    "../../scripts/requirements/trio.in",
+    "../../scripts/requirements/boto3.in",
+    "../../scripts/requirements/black.in",
+    "../../scripts/requirements/jupyter.in",
+];
+
+fn resolve_warm(c: &mut Criterion<WallTime>) {
+    for requirements in REQUIREMENTS {
+        let requirements = fs::canonicalize(requirements).unwrap();
+        let name = format!(
+            "resolve_warm_{}",
+            requirements.file_stem().unwrap().to_string_lossy()
+        );
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let output_file = Path::new("requirements.txt");
+        let cache_dir = Path::new(".cache");
+
+        c.bench_function(&name, |b| {
+            b.iter_batched(
+                || fs::remove_file(output_file).ok(),
+                |_| {
+                    let mut command = Command::new("uv");
+                    command
+                        .args(["pip", "compile"])
+                        .arg(&requirements)
+                        .arg("--cache-dir")
+                        .arg(cache_dir)
+                        .arg("--output-file")
+                        .arg(output_file)
+                        .current_dir(&temp_dir)
+                        .stdout(Stdio::null())
+                        .stderr(Stdio::null());
+
+                    command.status().unwrap();
+                },
+                BatchSize::SmallInput,
+            );
+        });
+    }
+}
+
+fn install_warm(c: &mut Criterion<WallTime>) {
+    for requirements in REQUIREMENTS {
+        let requirements = fs::canonicalize(requirements).unwrap();
+        let name = format!(
+            "install_warm_{}",
+            requirements.file_stem().unwrap().to_string_lossy()
+        );
+
+        let venv_dir = Path::new(".venv");
+        env::set_var("VIRTUAL_ENV", venv_dir);
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let cache_dir = Path::new("./.cache");
+
+        c.bench_function(&name, |b| {
+            b.iter_batched(
+                || {
+                    let mut command = Command::new("virtualenv");
+                    command
+                        .args(["--clear", "-p", "3.12"])
```

Comment:
We can probably use `uv venv` here instead of `virtualenv`, just to remove the external dependency.

Reply:
I think we use `virtualenv` in the benchmark script just because that's designed to benchmark uv against other tools, and so it removes one source of (possible) variance.